### PR TITLE
ErrorView: Better detection of no-data responses

### PR DESCRIPTION
**Pull request by leeoniya**
Fixes https://github.com/grafana/support-escalations/issues/5218

@torkelo was there a specific reason for only checking when `data.series.length === 1`?

### CODE REVIEW

File: public/app/features/panel/components/PanelDataErrorView.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ function getMessageFor(
   }
 
   // In some cases there is a data frame but with no fields
```
Comment (nmarrs) [inline public/app/features/panel/components/PanelDataErrorView.tsx:69]: We may also want to update / remove this comment as new logic doesn't match 🤔

**Review by nmarrs — APPROVED** [review]
I think updated check makes sense

**Review by torkelo — APPROVED** [review]
No reason that I can remember :)

**Review by bohandley — APPROVED** [review]
LGTM, thank you @leeoniya!

**Pull request merged**